ipfx/dataset/ephys_data_set.py

```diff
@@ -33,57 +46,79 @@ class EphysDataSet(object):
     VOLTAGE_CLAMP = "VoltageClamp"
     CURRENT_CLAMP = "CurrentClamp"
 
-    def __init__(self,
-                 data: EphysDataInterface,
-                 sweep_info: Optional[Dict[str, Any]] = None):
-
-        self.data = data
-        self.ontology = data.ontology
-
-        if sweep_info is None:
-            sweep_info = self.extract_sweep_info()
+    @property
+    def ontology(self) -> StimulusOntology:
+        """The stimulus ontology maps short-form stimulus codes to complete
+        descriptions.
+        """
+        return self._data.ontology
 
-        self.build_sweep_table(sweep_info)
+    @property
+    def sweep_table(self) -> pd.DataFrame:
+        """Each row of the sweep table contains the metadata for a single 
+        sweep. In particular details of the stimulus presented and the clamp 
+        mode. See EphysDataInterface.get_sweep_metadata for more information.
 
-    def extract_sweep_info(self) -> Dict[str,Any]:
         """
-        Extract sweep information for all sweeps
+        if not hasattr(self, "_sweep_table"):
+            self._sweep_table = pd.DataFrame([
+                self._data.get_sweep_metadata(num)
+                for num in self._data.sweep_numbers
+            ])
+        return self._sweep_table
+
+    def __init__(
+            self,
+            data: EphysDataInterface
+    ):
+        """EphysDataSet is the preferred interface for running analyses or 
+        pipeline code.
 
-        Returns
-        -------
-        sweep information
+        Parameters
+        ----------
+        data : This object must implement the EphysDataInterface. It will 
+            handle any loading of data from external sources (such as NWB2 
+            files)
         """
+        self._data: EphysDataInterface = data
 
-        sweep_info = []
-
-        for index, sweep_map in self.data.sweep_map_table.iterrows():
-            sweep_num = sweep_map["sweep_number"]
-            sweep_info.append(self.data.extract_sweep_record(sweep_num))
+    def _setup_stimulus_repeat_lookup(self):
+        """Construct a mapping from sweep number to the number of presentations 
+        of the stimulus presented during that sweep before and during that 
```

Comment:
this is a confusing docstring "presented during that sweep before and during that..."